File: Vjudge Contest/E_solve.py
def can_sort_by_reversing_segment():
    import sys
    input = sys.stdin.read
    data = input().split()
    
 
    n = int(data[0])
    array = list(map(int, data[1:]))
    

    left = 0
    while left < n - 1 and array[left] <= array[left + 1]:
        left += 1
    

    if left == n - 1:
        print("yes")
        print(1, 1)
        return
    

    right = n - 1
    while right > 0 and array[right] >= array[right - 1]:
        right -= 1

    array[left:right + 1] = reversed(array[left:right + 1])
    
    if all(array[i] <= array[i + 1] for i in range(n - 1)):
        print("yes")
        print(left + 1, right + 1)  

can_sort_by_reversing_segment()

File: Vjudge Contest/E_solve.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

void canSortByReversingSegment() {
    int n;
    cin >> n; 
    vector<int> array(n);
    

    for (int i = 0; i < n; i++) {
        cin >> array[i];
    }

    int left = 0;
    while (left < n - 1 && array[left] <= array[left + 1]) {
        left++;
    }


    if (left == n - 1) {
        cout << "yes" << endl;
        cout << 1 << " " << 1 << endl; 
        return;
    }


    int right = n - 1;
    while (right > 0 && array[right] >= array[right - 1]) {
        right--;
    }

    reverse(array.begin() + left, array.begin() + right + 1);


    bool isSorted = true;
    for (int i = 0; i < n - 1; i++) {
        if (array[i] > array[i + 1]) {
            isSorted = false;
            break;
        }
    }

    if (isSorted) {
        cout << "yes" << endl;
        cout << left + 1 << " " << right + 1 << endl; // Convert to 1-based indexing
    } else {
        cout << "no" << endl;
    }
}

int main() {
    canSortByReversingSegment();
    return 0;
}
